### PR TITLE
merge fluid dist tests

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/PaddlePaddle/Paddle/issues/8571
Fix https://github.com/PaddlePaddle/Paddle/issues/8598

### CODE REVIEW

File: python/paddle/fluid/tests/book/test_fit_a_line.py [hunk 1]
```diff
@@ -42,27 +43,57 @@ def train(use_cuda, save_dirname):
...
+                    return
+                if math.isnan(float(avg_loss_value)):
+                    sys.exit("got NaN loss, training failed.")
+        raise AssertionError("Fit a line cost is too large, {0:2.2}".format(
+            avg_loss_value[0]))
+
+    if is_local:
+        train_loop(fluid.default_main_program())
+    else:
+        port = os.getenv("PADDLE_INIT_PORT", "6174")
```
Comment (gongweibao) [inline python/paddle/fluid/tests/book/test_fit_a_line.py]: It's hard code here.
There is a reference:
https://github.com/gongweibao/cloud/blob/2df615851d0191677e00b1027120baeffbc2174d/docker/paddle_k8s#L70
https://github.com/gongweibao/Paddle/blob/ffe213950d3b08eaf5f8119422662e554a9c7725/benchmark/cluster/vgg16/vgg16_fluid.py#L240

Comment (typhoonzero) [reply python/paddle/fluid/tests/book/test_fit_a_line.py]: Thought both ways are OK, `PADDLE_INIT_PORT` is also used in v2 when calling `paddle.init()`

File: python/paddle/fluid/tests/book/test_fit_a_line.py [hunk 2]
```diff
@@ -42,27 +43,57 @@ def train(use_cuda, save_dirname):
...
+                    if save_dirname is not None:
+                        fluid.io.save_inference_model(save_dirname, ['x'],
+                                                      [y_predict], exe)
+                    return
+                if math.isnan(float(avg_loss_value)):
+                    sys.exit("got NaN loss, training failed.")
+        raise AssertionError("Fit a line cost is too large, {0:2.2}".format(
+            avg_loss_value[0]))
+
+    if is_local:
```
Comment (Yancey1989) [inline python/paddle/fluid/tests/book/test_fit_a_line.py]: Is `is_local` necessary? We can run the local mode with the code under `book` directory.

Comment (Yancey1989) [reply python/paddle/fluid/tests/book/test_fit_a_line.py]: Got it, this PR merge book and book_dist folder.

**Pull request merged**